### ISSUE TITLE
Delay for first retry always zero

### ISSUE THREAD

**Issue by jeffreyhuey** [softonic/axios-retry#293]
I'm using axios-retry 4.5.0.  It seems no matter what retryDelay I use, linear, exponentialDelay or custom delay, the delay for the first retry is always zero (it's retried immediately after a failed http request).  Code like below.   Wonder if anybody else experiencing this situation? Thanks.
// Initialize axios-retry with the axios instance
axiosRetry(axios, {
  retries: 3, // Number of retry attempts
  retryDelay: (retryNumber) => {return Math.pow(2, retryNumber) * 2000 },
  retryCondition: (error) => {
    // Retry on network errors or 5xx status codes
    return axiosRetry.isNetworkError(error) || axiosRetry.isRetryableError(error);
  },
  onRetry: (retryCount) => {
     var now = new Date(); 
    // Extract the current time 
    var hours = now.getHours(); 
    var minutes = now.getMinutes(); 
    var seconds = now.getSeconds(); 
    // Format the time as a string 
    var currentTime = `${hours}:${minutes}:${seconds}`;
    console.log(currentTime + ` : retry attempt #${retryCount}`);
  }, 
  resetTimeout: true
});

Output of test program is like this: 
22:13:6 : http get sent
22:13:6 : retry attempt #1   <-----------------  First retry is immediate !!!
22:13:10 : retry attempt #2
22:13:18 : retry attempt #3
Error fetching data:

**Comment by ayyagari** [comment]
Yes I noticed the same thing. First retry happens with few tens of milliseconds right after the first call with v4.5.0.

**Comment by igorsantos07** [comment]
For some odd reason, I used `retryCount` instead of `retries`. That caused instant failure, calling `onMaxRetryTimesExceeded()` 🤡 

Took me quite a while to realize it seems I was rewriting what should be a protected field.

Maybe try commenting all config properties and uncommenting one by one, until you find the offender.
